components/places/src/storage/bookmarks.rs

```diff
@@ -426,6 +431,110 @@ impl Default for UpdateTreeLocation {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct ReadonlyBookmark {
+    pub guid: SyncGuid,
+    pub parent_guid: Option<SyncGuid>,
+    // Always 0 if parent_guid is None
+    pub position: BookmarkPosition,
```

Comment:
I think this should probably just be an int - `BookmarkPosition::Append` wouldn't make sense here. Ditto in the structs below.